docs/frontend-system/architecture/08-naming-patterns.md

```diff
@@ -0,0 +1,139 @@
+---
+id: naming-patterns
+title: Frontend System Naming Patterns
+sidebar_label: Naming Patterns
+# prettier-ignore
+description: Naming patterns in the frontend system
+---
+
+> **NOTE: The new frontend system is in a highly experimental phase**
+
+These are the naming patterns to adhere to within the frontend system. They help us keep exports and IDs consistent across packages and make it easier to understand the usage and intent of exports and IDs.
+
+As a rule, all names should be camel case, with the exceptions of plugin and extension IDs, which should use kebab case.
+
+### Plugins
+
+| Description | Pattern      | Examples                              |
+| ----------- | ------------ | ------------------------------------- |
+| ID          | `'<id>'`     | `'catalog'`, `'user-settings'`        |
+| Symbol      | `<id>Plugin` | `catalogPlugin`, `userSettingsPlugin` |
+
+Example:
+
+```ts
+export const userSettingsPlugin = createPlugin({
+  id: 'user-settings',
+  ...
+})
+```
+
+Note that while we use this naming pattern for the plugin instance this is only for internal usage within the package. Plugins are always exported as the default export of the plugin package.
+
+### Extensions
+
+| Description | Pattern                         | Examples                                                            |
+| ----------- | ------------------------------- | ------------------------------------------------------------------- |
+| Kind        | `<Kind>`                        | `Page`, `EntityCard`, `SearchResultListItem`                        |
```

Comment:
here it's a bit unclear what Kind is - it's not really _directly_ used anywhere, and in the ID example it has different casing, and you also pass it in as kebab case to the createExtension functions so ... maybe it's clearest to just remove this line?